src/OpenApi3/Generator/Client/ServerPluginGenerator.php

```diff
@@ -30,6 +30,13 @@ private function discoverServer(OpenApi $openApi): array
             if (\array_key_exists('host', $url)) {
                 $scheme = $url['scheme'] ?? 'https';
                 $baseUri = $scheme . '://' . trim($url['host'], '/');
+
+                $variables = $server->getVariables();
+
+                if ($variables['port']->getDefault()) {
```

Comment:
Please check is `port` variable exists and if content if different of `null` before accessing it

Reply:
And if `$server->getVariables()` is not null aswell